lib/init.c

```diff
@@ -296,31 +296,41 @@ void rpc_set_auxiliary_gids(struct rpc_context *rpc, uint32_t len, uint32_t* gid
 void rpc_set_error(struct rpc_context *rpc, const char *error_string, ...)
 {
         va_list ap;
-	char *old_error_string;
 
 #ifdef HAVE_MULTITHREADING
         if (rpc->multithreading_enabled) {
                 nfs_mt_mutex_lock(&rpc->rpc_mutex);
         }
 #endif /* HAVE_MULTITHREADING */
-        old_error_string = rpc->error_string;
+
         va_start(ap, error_string);
+	free(rpc->error_string);
```

Comment:
Ideally this whole function should just be the locking and then a call to the _locked() function
so that we only have one copy of the code that generates the new error string.

Reply:
yes, I thought too but due to the varargs we have to either take a va_list as argument in which case the locked() version cannot be directly called or we make that a macro. For now, I'll leave it as is.